crates/ruff_notebook/src/notebook.rs

```diff
@@ -80,14 +80,126 @@ impl Cell {
+        //
+        // For example, this is valid:
+        // ```jupyter
+        // cat /path/to/file
+        // cat /path/to/file
+        // ```
+        //
+        // But this is invalid:
+        // ```jupyter
+        // cat /path/to/file
+        // x = 1
+        // ```
+        //
+        // See: https://ipython.readthedocs.io/en/stable/interactive/magics.html
+        if line.split_whitespace().next().is_some_and(|token| {
+            matches!(
+                token,
+                "alias"
+                    | "alias_magic"
+                    | "autoawait"
+                    | "autocall"
+                    | "automagic"
+                    | "bookmark"
+                    | "cd"
+                    | "code_wrap"
+                    | "colors"
+                    | "conda"
+                    | "config"
+                    | "debug"
+                    | "dhist"
+                    | "dirs"
+                    | "doctest_mode"
+                    | "edit"
+                    | "env"
+                    | "gui"
+                    | "history"
+                    | "killbgscripts"
+                    | "load"
+                    | "load_ext"
+                    | "loadpy"
+                    | "logoff"
+                    | "logon"
+                    | "logstart"
+                    | "logstate"
+                    | "logstop"
+                    | "lsmagic"
+                    | "macro"
+                    | "magic"
+                    | "mamba"
+                    | "matplotlib"
+                    | "micromamba"
+                    | "notebook"
+                    | "page"
+                    | "pastebin"
+                    | "pdb"
+                    | "pdef"
+                    | "pdoc"
+                    | "pfile"
+                    | "pinfo"
+                    | "pinfo2"
+                    | "pip"
+                    | "popd"
+                    | "pprint"
+                    | "precision"
+                    | "prun"
+                    | "psearch"
+                    | "psource"
+                    | "pushd"
+                    | "pwd"
+                    | "pycat"
+                    | "pylab"
+                    | "quickref"
+                    | "recall"
+                    | "rehashx"
+                    | "reload_ext"
+                    | "rerun"
+                    | "reset"
+                    | "reset_selective"
+                    | "run"
+                    | "save"
+                    | "sc"
+                    | "set_env"
+                    | "sx"
+                    | "system"
+                    | "tb"
+                    | "time"
+                    | "timeit"
+                    | "unalias"
+                    | "unload_ext"
+                    | "who"
+                    | "who_ls"
+                    | "whos"
+                    | "xdel"
+                    | "xmode"
```

Comment:
Obviously some risk of this getting stale but I kind of doubt these change dramatically over time.

Reply:
I agree. There are user defined magics as well but I think that's out of scope 😅